src/application/service/authentication/EstonianIdAuthenticator.ts

```diff
@@ -0,0 +1,39 @@
+import { AuthenticationProvider } from '../../../domain/model/idAuthentication/AuthenticationProvider';
+import { GenerateAuthToken } from './GenerateAuthToken';
+import { GetExistingOrCreateNewUser } from '../users/GetExistingOrCreateNewUser';
+import { AuthenticationSessionToken, Authentication } from '../../../domain/model/idAuthentication/models';
+import { Authenticator, AuthCode } from '../../../domain/model/authentication/Authenticator';
+import { AuthenticationDetails } from '../../../domain/model/user/AuthenticationDetails';
+import { AuthenticationMethod } from '../../../domain/model/user/AuthenticationMethod';
+import { AuthenticationIdentifier } from '../../../domain/model/user/AuthenticationIdentifier';
+import { AuthenticationError } from './AuthenticationError';
+
+export class EstonianIdAuthenticator implements Authenticator {
```

Comment:
@kstam might be better to comment on architectural principles, but I'm a little surprised to see this class in the service layer.  I would have thought the service layer would just contain the generic authentication methods, and the specific implementation would occur lower down in the application stack.

Reply:
To be honest we didn't think hard about where they should be here. Happy for any suggestions